Sources/VaporJWT/Request+JWT.swift

```diff
@@ -0,0 +1,50 @@
+import HTTP
+import JWT
+
+extension Request {
+    /// Parses and returns a JWT from the request
+    /// if one exists.
+    /// The JWT will also be verified with the supplied
+    /// signer.
+    public func jwt(verifyUsing signer: Signer) throws -> JWT {
```

Comment:
For more flexibility we might want to decouple verification from retrieving `JWT`s from `Request`s. That way it is possible to do something with the `JWT` outside `LoginMiddleware` without having access to the signer and verifying multiple times unnecessarily. The method is called only once anyway, might as well verify at that call site.